pkg/pool-utils/contracts/BaseGeneralPool.sol

```diff
@@ -34,9 +34,8 @@ abstract contract BaseGeneralPool is IGeneralPool, BasePool {
         uint256[] memory balances,
         uint256 indexIn,
         uint256 indexOut
-    ) public virtual override onlyVault(swapRequest.poolId) returns (uint256) {
-        // Block all swaps when paused
-        _ensureNotPaused();
+    ) public override onlyVault(swapRequest.poolId) returns (uint256) {
```

Comment:
Maybe add a comment on the reason for removing virtual here (and in corresponding join/exit hooks). You're actually preventing derived contracts from overriding onSwap now, because doing so (and adding extra functions, as we did) would make the query functions and the live functions diverge.

Reply:
Yes, but I want to add that later once I fix queries + add tests.